lib/archive_importer/own_relayable_importer.rb

```diff
@@ -19,7 +21,9 @@ def fetch_parent(data)
         break entity_class::PARENT_TYPE if entity_class.const_defined?(:PARENT_TYPE)
       }
       entity = Diaspora::Federation::Mappings.model_class_for(type).find_by(guid: data.fetch(:parent_guid))
-      data[:parent] = Diaspora::Federation::Entities.related_entity(entity)
+      raise NoParentError if entity.nil?
```

Comment:
The same question as in https://github.com/diaspora/diaspora/pull/8010#discussion_r281341950 still exists. In theory this shouldn't happen, because in the validate step it already tries to fetch all the entities here

https://github.com/diaspora/diaspora/blob/f85f167f505985c6c34dc77f5c29493e28d7aa20/lib/archive_validator/own_relayable_validator.rb#L15-L17

using

https://github.com/diaspora/diaspora/blob/f85f167f505985c6c34dc77f5c29493e28d7aa20/lib/diaspora/federated/fetchable.rb#L9-L17

which already returns nil when not being able to be fetched, and if the parent is nil, it's not valid:

https://github.com/diaspora/diaspora/blob/a3196a108946eb331d007add064182e558e448aa/lib/archive_validator/relayable_validator.rb#L16-L18

which then should remove all invalid entities during the validation phase, so they aren't tried to be actually imported:

https://github.com/diaspora/diaspora/blob/f85f167f505985c6c34dc77f5c29493e28d7aa20/lib/archive_validator/collection_validator.rb#L9-L13

If this doesn't work as expected, it also means a lot of the things are done twice, which isn't intended (and could also lead to longer runtimes). If you have steps to reproduce this and can debug why entities that don't exist still end up at this point and aren't filtered out earlier already, it would be good to know and should be fixed there instead.

Reply:
The part: find_by(guid: data.fetch(:**parent_guid**))  makes the problem here, not detected by the validator. 
But we already talked that it seems to be inconvenient to fetch everything in the validation part.